pubsub.go

```diff
@@ -268,3 +338,40 @@ func bootstrapPubsub(ctx context.Context, cr routing.ContentRouting, host p2phos
 
 	wg.Wait()
 }
+
+func (wg *watchGroup) add(ctx context.Context, outCh chan []byte) {
+	proxy := make(chan []byte, 1)
+	n := wg.n
+	wg.listeners[n] = proxy
+	wg.n++
+
+	go func() {
+		defer close(outCh)
+		defer func() {
+			wg.lk.Lock()
+			delete(wg.listeners, n)
+			//TODO: watchgroup GC?
```

Comment:
Might as well. Just `if len(wg.listeners) == 0 { /* lock and remove watch group */ }`.